### PR TITLE
Implemented prime number function

### PR DESCRIPTION
**Pull request by shravan-cc**
Implemented prime number function by making use of assert function

### CODE REVIEW

File: prime.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+function assert(bool, message) {
```
Comment (shripada) [inline prime.js]: we better name the param as `condition` than the `bool`

File: prime.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            if (num % i === 0) {
+                return false;
+            }
+        }
+        return true;
+    } 
+}
+
+assert(isprime(9)===true,"The given  number is not prime number");
+assert(isprime(11)===false,"The given  number is prime number");
```
Comment (shripada) [inline prime.js]: good attempt, add more assertions for various inputs like 0, 1, invalid input etc.

**Review by shripada — CHANGES_REQUESTED** [review]

File: prime.js (hunk 1 above)
Comment (shravan-cc) [reply prime.js]: Thank you for your feedback. I have addressed the requested changes in the latest commit.

File: prime.js (hunk 2 above)
Comment (shravan-cc) [reply prime.js]: Okay Sir! I will do the necassary changes.

File: prime.js [hunk 3]
```diff
@@ -1,23 +1,50 @@
...
-assert(isprime(9)===false,"The given  number is not prime number");
-assert(isprime(11)===true,"The given  number is prime number");
-assert.throws(()=>isprime("string"),Error);
-assert.throws(()=>isprime(),Error);
-assert.throws(()=>isprime(1),Error);
+assert(isprime(9) === false, "The given  number is not prime number");
+assert(isprime(11) === true, "The given  number is prime number");
+assert.throws(() => isprime("string"), Error);
+assert.throws(() => isprime(), Error);
+assert.throws(() => isprime(1), Error);
```
Comment (shripada) [inline prime.js]: Good attempt.

File: prime.js [hunk 4]
```diff
@@ -1,23 +1,50 @@
...
+assert.throws(() => isprime(-1), Error);
+assert.throws(() => isprime(4.87), Error);
+
+function generateprimeseries(count) {
```
Comment (shripada) [inline prime.js]: we need to use camel case consistenly.  `generatePrimeSeries` is correct one.

File: prime.js [hunk 5]
```diff
@@ -1,23 +1,50 @@
...
+  let i = 2;
+  let cnt = 0;
+  let array = [];
+  if (typeof count !== "number" || !Number.isInteger(count)) {
```
Comment (shripada) [inline prime.js]: replace this with assert itself.

File: prime.js [hunk 6]
```diff
@@ -1,23 +1,50 @@
...
+    throw new Error("Arguement must be a valid integer greater than 1");
+  }
+  assert(Number.isInteger(count), "Argument must be an integer");
```
Comment (shripada) [inline prime.js:36]: I think if this passes, the second assertion might not be needed.

File: prime.js [hunk 7]
```diff
@@ -1,23 +1,50 @@
...
+    if (isprime(i)) {
+      cnt++;
+      array.push(i);
+    }
+    i++;
+  }
+  return array;
+}
+//console.log(typeof([1,2,3,4,5]))
+assert.deepStrictEqual(
```
Comment (shripada) [inline prime.js]: need more tests

File: prime.js [hunk 8]
```diff
@@ -1,23 +1,50 @@
...
+  return true;
 }
 
-assert(isprime(9)===false,"The given  number is not prime number");
-assert(isprime(11)===true,"The given  number is prime number");
-assert.throws(()=>isprime("string"),Error);
-assert.throws(()=>isprime(),Error);
-assert.throws(()=>isprime(1),Error);
+assert(isprime(9) === false, "The given  number is not prime number");
+assert(isprime(11) === true, "The given  number is prime number");
```
Comment (shripada) [inline prime.js]: also add test for a large prime number

**Review by shripada — CHANGES_REQUESTED** [review]

File: prime.js (hunk 4 above)
Comment (shravan-cc) [reply prime.js]: I performed the necessary changes.

File: prime.js (hunk 5 above)
Comment (shravan-cc) [reply prime.js]: I performed the necessary changes.

File: prime.js (hunk 6 above)
Comment (shravan-cc) [reply prime.js:36]: I performed the necessary changes.

File: prime.js (hunk 7 above)
Comment (shravan-cc) [reply prime.js]: I performed the necessary changes.

File: prime.js (hunk 8 above)
Comment (shravan-cc) [reply prime.js]: I performed the necessary changes.

**Pull request merged**